woocommerce/woocommerce-ios
# Coupon: Fix issue encoding coupons

Closes: #7114


### Description

This PR fixes the encoder of coupons by adding the missing fields `individualUse` and `excludeSaleItems`. Unit test has also been added for extra confidence.

### Testing instructions

1. Go to Menu > Coupons > Select a coupon
2. Select the ellipsis button and choose Edit Coupon
3. Select Usage Restrictions and toggle the values for "Individual use" and "Excludes sale items" and then save the changes.
4. Open the coupon details again, notice that the new values are displayed on the coupon details screen and the edit form displays the correct values..

### Screenshots


https://user-images.githubusercontent.com/5533851/174758295-821299f7-637b-44de-97de-366e422a4220.mp4



---
